CMSgov/onemac
# OY2 4565 User Management Admin Dashboard

Story: https://qmacbis.atlassian.net/browse/OY2-4565
EndPoint: https://d9lvdy1y1ajsi.cloudfront.net

Back End Changes:
- followed the packageExists pattern to consolidate the getUser function into the utils directory, updated getUser and putUser accordingly
- created getMyUserList lambda for returning the appropriate userList data based on calling user type
  - broke out the different user types into separate files in the users directory similar to changeRequests
  - created getScanFor to return which user type the calling user type should use
  - created transformUserList to manipulate the User Item from dynamoDB into the list that the front end needs
  - created utils/getStatusDetails for common functionality in transformUserList
  - created utils/getAuthorizedStateList to get a user's list of authorized states
- updated loadTestUsers.sh, 
  - gave stateadmins only ONE state
  - adding pending status before the active for some stateadmins
  - added systemadmintest@cms.hhs.local user for testing system admin role
 
Front End Changes:
- created UserManagement page and added to Routes & Nav bar
- shared user pending check functionality pulled from Dashboard into libs/userLib.js
- blocked stateusers from page by re-routing to Dashboard (was causing data mishaps and Zach's story will fix)
- dynamically call getMyUserList and display what is returned in a table matching the Figma prototype
- per-row PopUp Menu shows the available actions based on user status
- clicking the PopUp Menu items calls the API for setUserStatus with doneBy, userEmail, and newStatus
- after API is called, Success Alert is shown and the user list is reloaded

NOTE: the setUserStatus API call is incomplete because of the lambda dependency
NOTE: content is missing for several confirmation messages and will be handled in wrapper stories
NOTE: Figma prototype shows a custom alert message including user's name -- converting the alert message to a dynamic message type is outside current scope.
NOTE: FirstName LastName is not yet being stored in User Table, so is not currently sent through user list, but it should be ready to connect.
NOTE: Back End sends responses about calling users’ access - not sure we have defined what should happen for revoked or denied callers at the front end.

To Test:
1. Login as any State User - click User Management link - should be redirected to Dashboard
2. Login as stateadminactiveMI@cms.hhs.local
    a. verify that only the State Users from MI are being shown
    b.Verify that the state column is NOT shown
    c. check the style against the Figma prototype
    d. Use the popup menus for each of the actions and verify that the "correct" messages are being shown.  Please note that some of the refined messages are missing.
4. check that other stateadmin statuses do not show any users
5. Login as cmsapproveractive@cms.hhs.com
   a. verify that all stateadmins are shown
   b. check that the status changes show "correct" messages
6. Final role to check is: systemadmintest@cms.hhs.local
  a. Verify that cmsapprover list is shown to systemadmin
  b. Verify the messaging for the status changes.
